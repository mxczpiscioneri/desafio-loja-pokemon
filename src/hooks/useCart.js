import { useState } from 'react'

export const useCart = () => {
  const [open, setOpen] = useState(false)

  const toggleDrawer = () => {
    setOpen(status => !status)
  }

  return {
    open,
    toggleDrawer
  }
}
