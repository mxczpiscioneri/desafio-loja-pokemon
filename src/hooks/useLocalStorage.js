import { useState } from 'react'

export const keys = { type: { slug: 'type', value: 'POKEMON@TYPE' } }

export const useLocalStorage = () => {
  const [storedValue, setStoredValue] = useState(() => {
    let stored = null
    try {
      const typeSaved = window.localStorage.getItem(keys.type.value) || 'all'
      stored = { type: typeSaved }
    } catch (error) { }
    return stored
  })

  const setValue = (key, value) => {
    try {
      setStoredValue({ [keys[key.slug].slug]: value })
      window.localStorage.setItem(key.value, value)
    } catch (error) { }
  }

  return {
    storedValue,
    setValue
  }
}
