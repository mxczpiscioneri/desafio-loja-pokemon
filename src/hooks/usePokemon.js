/* eslint complexity: ["error", 5] */

import { useState, useCallback } from 'react'
import PokeApiService from '../services/pokeapi'

export const usePokemon = () => {
  const limit = 24
  const [loading, setLoading] = useState(true)
  const [error, setError] = useState(false)
  const [data, setData] = useState([])
  const [offset, setOffset] = useState(0)

  const paginate = () => {
    setOffset(offsetCurrent => offsetCurrent + limit)
  }

  const getPokemonDetail = async (pokemons) => {
    const pokemonListDetails = await Promise.all(
      pokemons.map(async ({ url }) => {
        try {
          const responseDetails = await PokeApiService.getDetails(url)
          return responseDetails && responseDetails.data
        } catch (exception) {
          return {}
        }
      })
    )

    return pokemonListDetails
  }

  const getAllPokemon = useCallback(async () => {
    try {
      setError(false)
      setLoading(true)

      const responseList = await PokeApiService.getAll(offset, limit)
      let responseDetails = await getPokemonDetail(responseList.data.results)
      responseDetails = responseDetails.filter(value => Object.keys(value).length !== 0)

      setData(dataCurrent => dataCurrent.concat(responseDetails))
    } catch (err) {
      if (err.response && err.response.status === 404) {
        setData([])
      } else {
        setError(true)
      }
    } finally {
      setLoading(false)
    }
  }, [offset])

  return {
    loading,
    data,
    error,
    setError,
    offset,
    paginate,
    getAllPokemon
  }
}
