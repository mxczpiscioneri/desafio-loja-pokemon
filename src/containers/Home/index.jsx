/* eslint-disable no-unused-expressions */
/* eslint complexity: ["error", 5] */

import React, { useEffect } from 'react'
import Alert from '@material-ui/lab/Alert'
import { Button, CardActionArea, Container, Grid, Snackbar, Typography } from '@material-ui/core'
import Loading from '../../components/Loading'
import { usePokemon } from '../../hooks/usePokemon'
import { GridStyled, CardStyled, ImageStyled, CardContentStyled, NameStyled, TypesStyled, PriceStyled, AvatarStyled } from './style'

const Home = () => {
  const {
    loading,
    data,
    error,
    setError,
    paginate,
    getAllPokemon
  } = usePokemon()

  useEffect(() => {
    getAllPokemon()
  }, [getAllPokemon])

  return (
    <Container>
      {loading &&
        <Loading backdrop={true} />
      }
      {error &&
        <Snackbar
          open={error}
          anchorOrigin={{ horizontal: 'right', vertical: 'top' }}
          autoHideDuration={6000}
          onClose={() => setError(false)}
        >
          <Alert variant="filled" severity="error">
            Ocorreu um erro, tente novamente!
          </Alert>
        </Snackbar>
      }
      <Grid container spacing={2}>
        {data.length > 0 ? (
          <>
            {data.map(item => (
              <GridStyled item xs={6} sm={4} md={3} lg={2} key={item.id}>
                <CardStyled>
                  <CardActionArea>
                    <ImageStyled
                      alt={item.name}
                      title={item.name}
                      src={item.sprites.front_default}
                      onMouseOver={(e) => {
                        e.currentTarget.src = item.sprites.back_default
                      }}
                      onMouseOut={(e) => {
                        e.currentTarget.src = item.sprites.front_default
                      }}
                    />
                    <CardContentStyled>
                      <NameStyled variant="h5">
                        {item.name}
                      </NameStyled>
                      <PriceStyled variant="subtitle1">
                        <b>$</b> {item.base_experience}
                      </PriceStyled>
                      <TypesStyled>
                        {item.types.map(type => (
                          <AvatarStyled
                            key={type.type.name}
                            src={require(`../../assets/icons/${type.type.name}.svg`).default}
                            alt={type.type.name}
                          />
                        ))}
                      </TypesStyled>
                    </CardContentStyled>
                  </CardActionArea>
                </CardStyled>
              </GridStyled>
            ))}
            <Grid item xs={12} align="center">
              <Button data-testid="btnLoadMore" variant="outlined" size="large" color="primary" onClick={() => paginate()}>
              Carregar mais
              </Button>
            </Grid>
          </>
        ) :
          !loading && (
            <Grid item xs={12} align="center">
              <Typography>Nenhum Pokemon encontrado...</Typography>
            </Grid>
          )
        }

      </Grid>
    </Container>
  )
}

export default Home
