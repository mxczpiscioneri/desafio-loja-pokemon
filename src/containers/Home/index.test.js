import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import { fireEvent, render, screen } from '@testing-library/react'
import Home from '.'
import { usePokemon } from '../../hooks/usePokemon'
import mockDetailsPokemon from '../../assets/mock-data/details-pokemon.json'

jest.mock('../../hooks/usePokemon')

const containerComponent = (
  <BrowserRouter>
    <Home />
  </BrowserRouter>
)

describe('Home container', () => {
  test('Render layout', () => {
    const mockHooks = {
      loading: true,
      data: [mockDetailsPokemon],
      error: false,
      setError: jest.fn(),
      offset: 0,
      paginate: jest.fn(),
      getAllPokemon: jest.fn()
    }
    usePokemon.mockReturnValue(mockHooks)

    const component = render(containerComponent)

    fireEvent.mouseOver(screen.getByAltText('bulbasaur'))
    fireEvent.mouseOut(screen.getByAltText('bulbasaur'))

    expect(component).toMatchSnapshot()
  })

  test('Render without pokemon', () => {
    const mockHooks = {
      loading: false,
      data: [],
      error: false,
      setError: jest.fn(),
      offset: 0,
      paginate: jest.fn(),
      getAllPokemon: jest.fn()
    }
    usePokemon.mockReturnValue(mockHooks)

    const component = render(containerComponent)
    expect(component).toMatchSnapshot()
  })

  test('Render with error', () => {
    const mockHooks = {
      loading: false,
      data: [],
      error: true,
      setError: jest.fn(),
      offset: 0,
      paginate: jest.fn(),
      getAllPokemon: jest.fn()
    }
    usePokemon.mockReturnValue(mockHooks)

    const component = render(containerComponent)

    expect(component).toMatchSnapshot()
  })
})
