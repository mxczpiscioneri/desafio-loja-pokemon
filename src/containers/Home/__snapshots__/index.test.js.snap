// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home container Render layout 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg"
      >
        <div
          aria-hidden="true"
          class="MuiBackdrop-root sc-bdfBwQ iodWSc"
          style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        >
          <div
            class="sc-gsTCUz gomrDe"
          >
            <div
              class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
              role="progressbar"
              style="width: 40px; height: 40px;"
            >
              <svg
                class="MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                  cx="44"
                  cy="44"
                  fill="none"
                  r="20.2"
                  stroke-width="3.6"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root sc-dlfnbm hbZnVW MuiGrid-item MuiGrid-grid-xs-2"
          >
            <div
              class="MuiPaper-root MuiCard-root sc-hKgILt kmFPbB MuiPaper-elevation1 MuiPaper-rounded"
            >
              <button
                class="MuiButtonBase-root MuiCardActionArea-root"
                tabindex="0"
                type="button"
              >
                <img
                  alt="bulbasaur"
                  class="sc-eCssSg fFPGzd"
                  src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
                  title="bulbasaur"
                />
                <div
                  class="MuiCardContent-root sc-jSgupP cuDwNq"
                >
                  <h5
                    class="MuiTypography-root sc-gKsewC hhnpme MuiTypography-h5"
                  >
                    bulbasaur
                  </h5>
                  <h6
                    class="MuiTypography-root sc-pFZIQ hZisWE MuiTypography-subtitle1"
                  >
                    <b>
                      $
                    </b>
                     
                    64
                  </h6>
                  <div
                    class="MuiBox-root MuiBox-root-1 sc-iBPRYJ kEDcat"
                  >
                    <div
                      class="MuiAvatar-root MuiAvatar-circle sc-fubCfw jeiXPA"
                    >
                      <img
                        alt="grass"
                        class="MuiAvatar-img"
                        src="grass.svg"
                      />
                    </div>
                    <div
                      class="MuiAvatar-root MuiAvatar-circle sc-fubCfw jeiXPA"
                    >
                      <img
                        alt="poison"
                        class="MuiAvatar-img"
                        src="poison.svg"
                      />
                    </div>
                  </div>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight"
                />
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            align="center"
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-outlinedSizeLarge MuiButton-sizeLarge"
              data-testid="btnLoadMore"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Carregar mais
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root sc-bdfBwQ iodWSc"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <div
          class="sc-gsTCUz gomrDe"
        >
          <div
            class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
            role="progressbar"
            style="width: 40px; height: 40px;"
          >
            <svg
              class="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
      >
        <div
          class="MuiGrid-root sc-dlfnbm hbZnVW MuiGrid-item MuiGrid-grid-xs-2"
        >
          <div
            class="MuiPaper-root MuiCard-root sc-hKgILt kmFPbB MuiPaper-elevation1 MuiPaper-rounded"
          >
            <button
              class="MuiButtonBase-root MuiCardActionArea-root"
              tabindex="0"
              type="button"
            >
              <img
                alt="bulbasaur"
                class="sc-eCssSg fFPGzd"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
                title="bulbasaur"
              />
              <div
                class="MuiCardContent-root sc-jSgupP cuDwNq"
              >
                <h5
                  class="MuiTypography-root sc-gKsewC hhnpme MuiTypography-h5"
                >
                  bulbasaur
                </h5>
                <h6
                  class="MuiTypography-root sc-pFZIQ hZisWE MuiTypography-subtitle1"
                >
                  <b>
                    $
                  </b>
                   
                  64
                </h6>
                <div
                  class="MuiBox-root MuiBox-root-1 sc-iBPRYJ kEDcat"
                >
                  <div
                    class="MuiAvatar-root MuiAvatar-circle sc-fubCfw jeiXPA"
                  >
                    <img
                      alt="grass"
                      class="MuiAvatar-img"
                      src="grass.svg"
                    />
                  </div>
                  <div
                    class="MuiAvatar-root MuiAvatar-circle sc-fubCfw jeiXPA"
                  >
                    <img
                      alt="poison"
                      class="MuiAvatar-img"
                      src="poison.svg"
                    />
                  </div>
                </div>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          align="center"
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-outlinedSizeLarge MuiButton-sizeLarge"
            data-testid="btnLoadMore"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Carregar mais
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Home container Render with error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg"
      >
        <div
          class="MuiSnackbar-root MuiSnackbar-anchorOriginTopRight"
        >
          <div
            class="MuiPaper-root MuiAlert-root MuiAlert-filledError MuiPaper-elevation0"
            direction="down"
            role="alert"
            style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
          >
            <div
              class="MuiAlert-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
            </div>
            <div
              class="MuiAlert-message"
            >
              Ocorreu um erro, tente novamente!
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            align="center"
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Nenhum Pokemon encontrado...
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <div
        class="MuiSnackbar-root MuiSnackbar-anchorOriginTopRight"
      >
        <div
          class="MuiPaper-root MuiAlert-root MuiAlert-filledError MuiPaper-elevation0"
          direction="down"
          role="alert"
          style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        >
          <div
            class="MuiAlert-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
          </div>
          <div
            class="MuiAlert-message"
          >
            Ocorreu um erro, tente novamente!
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
      >
        <div
          align="center"
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Nenhum Pokemon encontrado...
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Home container Render without pokemon 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            align="center"
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Nenhum Pokemon encontrado...
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
      >
        <div
          align="center"
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Nenhum Pokemon encontrado...
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
